Python/codecs.c

```diff
@@ -50,6 +50,27 @@ int PyCodec_Register(PyObject *search_function)
     return -1;
 }
 
+int
+PyCodec_Unregister(PyObject *search_function)
+{
+    PyInterpreterState *interp = PyInterpreterState_Get();
+    PyObject *codec_search_path = interp->codec_search_path;
+    /* Do nothing if codec_search_path is not created yet or was cleared. */
+    if (codec_search_path == NULL) {
+        return 0;
+    }
+
+    Py_ssize_t n = PyList_Size(codec_search_path);
+    for (Py_ssize_t i = 0; i < n; i++) {
+        PyObject *item = PyList_GetItem(codec_search_path, i);
+        if (item == search_function) {
+            PyDict_Clear(interp->codec_search_cache);
```

Comment:
Hum, would you mind to add tests on codec_search_cache, just in case?

```suggestion
            if (interp->codec_search_cache != NULL) {
                assert(PyDict_Check(interp->codec_search_cache));
                PyDict_Clear(interp->codec_search_cache);
            }
```

Reply:
no probleam, I update it now

Reply:
Do we need add this line `assert(PyDict_Check(interp->codec_search_cache));`? 
Looks like the `PyDict_Clear` have checked the op is dict or not.

Reply:
Calling PyDict_Clear(op) on a type other than dict does nothing which would be a bug.

By the way, I would even prefer `assert(PyDict_CheckExact(interp->codec_search_cache));`.

Reply:
make sense, I will update it soon.